File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/dialogue/port/ConversationRepository.java
package com.study.webflux.rag.domain.dialogue.port;

import com.study.webflux.rag.domain.dialogue.model.ConversationSessionId;
import com.study.webflux.rag.domain.dialogue.model.ConversationTurn;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface ConversationRepository {

	Mono<ConversationTurn> save(ConversationTurn turn);

	Flux<ConversationTurn> findRecent(ConversationSessionId sessionId, int limit);

	Flux<ConversationTurn> findAll(ConversationSessionId sessionId);
}

File: backend/domain/src/main/kotlin/com/miyou/app/domain/dialogue/port/ConversationRepository.kt
package com.miyou.app.domain.dialogue.port

import com.miyou.app.domain.dialogue.model.ConversationSessionId
import com.miyou.app.domain.dialogue.model.ConversationTurn
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface ConversationRepository {
    fun save(turn: ConversationTurn): Mono<ConversationTurn>

    fun findRecent(
        sessionId: ConversationSessionId,
        limit: Int,
    ): Flux<ConversationTurn>
}
